File: hackertracker/src/main/java/com/shortstack/hackertracker/BottomSheet/ReviewBottomSheetDialogFragment.java
package com.shortstack.hackertracker.BottomSheet;

import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomSheetDialogFragment;
import android.view.View;

import com.github.stkent.amplify.prompt.DefaultLayoutPromptView;
import com.github.stkent.amplify.prompt.interfaces.IPromptPresenter;
import com.github.stkent.amplify.tracking.interfaces.IEvent;
import com.github.stkent.amplify.tracking.interfaces.IEventListener;
import com.orhanobut.logger.Logger;
import com.shortstack.hackertracker.R;

import butterknife.BindView;
import butterknife.ButterKnife;

public class ReviewBottomSheetDialogFragment extends BottomSheetDialogFragment {

    @BindView(R.id.prompt_view)
    DefaultLayoutPromptView promptView;

    public static ReviewBottomSheetDialogFragment newInstance() {
        Bundle args = new Bundle();

        ReviewBottomSheetDialogFragment fragment = new ReviewBottomSheetDialogFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void setupDialog(Dialog dialog, int style) {
        super.setupDialog(dialog, style);
        View view = View.inflate(getContext(), R.layout.bottom_sheet_review, null);
        dialog.setContentView(view);
        ButterKnife.bind(this, view);
        promptView.getPresenter().start();
        promptView.getPresenter().addPromptEventListener(new IPromptPresenter() {
            @Override
            public void addPromptEventListener(@NonNull IEventListener promptEventListener) {

            }

            @Override
            public void start() {

            }

            @Override
            public void reportUserOpinion(@NonNull UserOpinion userOpinion) {
            }

            @Override
            public void reportUserFeedbackAction(@NonNull UserFeedbackAction userFeedbackAction) {
            }

            @NonNull
            @Override
            public Bundle generateStateBundle() {
                return null;
            }

            @Override
            public void restoreStateFromBundle(@NonNull Bundle bundle) {
            }

            @Override
            public void notifyEventTriggered(@NonNull IEvent event) {
                Logger.d("Review event triggered: " + event.getTrackingKey());
                if (event.getTrackingKey().equals("PROMPT_DISMISSED") || event.getTrackingKey().equals("THANKS_SHOWN")) {
                    dismiss();
                }
            }
        });
    }
}

File: hackertracker/src/main/java/com/shortstack/hackertracker/BottomSheet/ReviewBottomSheetDialogFragment.kt
package com.shortstack.hackertracker.BottomSheet

import android.app.Dialog
import android.os.Bundle
import android.support.design.widget.BottomSheetDialogFragment
import android.view.View
import com.github.stkent.amplify.prompt.interfaces.IPromptPresenter
import com.github.stkent.amplify.tracking.interfaces.IEvent
import com.github.stkent.amplify.tracking.interfaces.IEventListener
import com.orhanobut.logger.Logger
import com.shortstack.hackertracker.R
import kotlinx.android.synthetic.main.bottom_sheet_review.view.*

class ReviewBottomSheetDialogFragment : BottomSheetDialogFragment() {

    override fun setupDialog(dialog: Dialog, style: Int) {
        super.setupDialog(dialog, style)
        val view = View.inflate(context, R.layout.bottom_sheet_review, null)
        dialog.setContentView(view)

        view.prompt_view!!.presenter.start()
        view.prompt_view!!.presenter.addPromptEventListener(object : IPromptPresenter {
            override fun addPromptEventListener(promptEventListener: IEventListener) {

            }

            override fun start() {

            }

            override fun reportUserOpinion(userOpinion: IPromptPresenter.UserOpinion) {
                // TODO: Add analytics for user feedback.
            }

            override fun reportUserFeedbackAction(userFeedbackAction: IPromptPresenter.UserFeedbackAction) {
                // TODO: Add analytics for user actions.
            }

            override fun generateStateBundle(): Bundle {
                return Bundle()
            }

            override fun restoreStateFromBundle(bundle: Bundle) {}

            override fun notifyEventTriggered(event: IEvent) {
                Logger.d("Review event triggered: " + event.trackingKey)
                if (event.trackingKey == "PROMPT_DISMISSED" || event.trackingKey == "THANKS_SHOWN") {
                    dismiss()
                }
            }
        })
    }

    companion object {

        fun newInstance(): ReviewBottomSheetDialogFragment {
            val args = Bundle()

            val fragment = ReviewBottomSheetDialogFragment()
            fragment.arguments = args
            return fragment
        }
    }
}
